services/gitea/gitea-pull-requests.tester.js

```diff
@@ -0,0 +1,173 @@
+    message: isMetricOpenIssues,
+  })
+
+t.create('Open pulls by label (raw)')
+  .get(
+    '/open-raw/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org&labels=upstream',
+  )
+  .expectBadge({
+    label: 'open upstream pull requests',
+    message: isMetric,
+  })
+
+t.create('Opened pulls by Scoped label')
+  .get(
+    '/open/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org&labels=failure/new',
+  )
+  .expectBadge({
+    label: 'failure/new pull requests',
+    message: isMetricOpenIssues,
+  })
+
+/**
+ *  Closed pulls number case
+ */
+t.create('Closed pulls')
+  .get(
+    '/closed/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org',
+  )
+  .expectBadge({
+    label: 'pull requests',
+    message: isMetricClosedIssues,
+  })
+
+t.create('Closed pulls raw')
+  .get(
+    '/closed-raw/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org',
+  )
+  .expectBadge({
+    label: 'closed pull requests',
+    message: isMetric,
+  })
+
+t.create('Closed pulls by label is > zero')
+  .get(
+    '/closed/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org&labels=bug',
+  )
+  .expectBadge({
+    label: 'bug pull requests',
+    message: isMetricClosedIssues,
+  })
+
+t.create('Closed pulls by  multi-word label is > zero')
+  .get(
+    '/closed/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org&labels=bug,good%20first%20issue',
+  )
+  .expectBadge({
+    label: 'bug,good first issue pull requests',
+    message: isMetricClosedIssues,
+  })
+
+t.create('Closed pulls by label (raw)')
+  .get(
+    '/closed-raw/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org&labels=bug',
+  )
+  .expectBadge({
+    label: 'closed bug pull requests',
+    message: isMetric,
+  })
+
+/**
+ *  All pulls number case
+ */
+t.create('All pulls')
+  .get('/all/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org')
+  .expectBadge({
+    label: 'pull requests',
+    message: Joi.string().regex(
+      /^([0-9]+[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY]) all$/,
+    ),
+  })
+
+t.create('All pulls raw')
+  .get(
+    '/all-raw/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org',
+  )
+  .expectBadge({
+    label: 'all pull requests',
+    message: isMetric,
+  })
+
+t.create('All pulls by label is > zero')
+  .get(
+    '/all/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org&labels=upstream',
+  )
+  .expectBadge({
+    label: 'upstream pull requests',
+    message: Joi.string().regex(
+      /^([0-9]+[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY]) all$/,
+    ),
+  })
+
+t.create('All pulls by multi-word label is > zero')
+  .get(
+    '/all/CanisHelix/shields-badge-test.json?gitea_url=https://codeberg.org&labels=question,enhancement',
+  )
+  .expectBadge({
+    label: 'question,enhancement pull requests',
+    message: Joi.string().regex(
+      /^([0-9]+[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY]) all$/,
```

Comment:
Why the custom validators here?

Reply:
This was taken from the Gitlab tests, presumably it was used there due to the lack of an **isMetricAllIssues**. There are validators for **isMetricOpenIssues** and **isMetricClosedIssues** but not one for an All.

Is using the function `isMetricWithPattern(/ all/)` okay? Or would it be preferred to add a new **isMetricAllIssues** to _test-validators.js_, that way other tests can benefit too?

Reply:
Can we use a repo that has >0 forks for the test
e.g: https://codeberg.org/Codeberg/Documentation